examples/maps/main.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+
+// Tiles are 256x256px wide.
+// Url is https://tile.openstreetmap.org/{zoom}/{x}/{y}.png
+// zoom starts at 1.
+// x and y go from 0 to 2^zoom - 1.
+
+use core::cell::RefCell;
+use core::future::Future;
+use core::pin::Pin;
+use core::task::{Context, Poll};
+use slint::{Rgba8Pixel, SharedPixelBuffer, VecModel};
+use std::collections::BTreeMap;
+use std::rc::Rc;
+
+const TILE_SIZE: isize = 256;
+
+slint::slint! {
```

Comment:
Overall it looks good to me. I would prefer if we work file based for the Slint code also for the examples. For me it's part of best practices but it's not mandatory right now for this example.

Reply:
I made it this way because this emphasis on simplicity as in the whole thing (Rust + Slint) fits in a single file of less than 500 lines

Reply:
Yes sure. I only want to give developers an example as best practices to work file based. Because the macro is in real live applications not handy. But it's my understanding ;-)